### PR TITLE
Allow customization of widget list

### PR DESCRIPTION
**Pull request by lslachciak**
[Issue] https://github.com/Samsung/TAU/issues/1372
[Problem] No possibility to customize widget list using add window
[Solution] Update activeWebClipList according to user choice

Signed-off-by: Lukasz Slachciak <l.slachciak@samsung.com>

### CODE REVIEW

**Comment by lslachciak** [comment]
Now user can customize widget list
image

File: examples/mobile/HomeApp/js/app.js [hunk 1]
```diff
@@ -75,8 +75,21 @@
...
 
 	function onPopupSubmit() {
-		getWidgetsState();
-		toggleWidgetsState();
+
+		activeWebClipList = [];
+		webClipList.forEach(function (webclip) {
+			var webClipName = webclip;
+
+			webClipName = webClipName.replace("webclip/", "")
```
Comment (HunseopJeong) [inline examples/mobile/HomeApp/js/app.js]: Need to add a semicolon.

File: examples/mobile/HomeApp/js/app.js [hunk 2]
```diff
@@ -112,6 +125,58 @@
 		tau.engine.createWidgets(webclipsContainer);
 	}
 
+	function updateWebClipListPopup() {
+
+		var popupList = document.getElementById("popup-list");
+		//console.log("current: " + current);
```
Comment (HunseopJeong) [inline examples/mobile/HomeApp/js/app.js]: Remove the comment

File: examples/mobile/HomeApp/js/app.js [hunk 3]
```diff
@@ -112,6 +125,58 @@
...
+			popupList.removeChild(li);
+		});
+
+		webClipList.forEach(function (webclip) {
+			var li = document.createElement("li"),
+				input = document.createElement("input"),
+				label = document.createElement("label"),
+				webClipName = webclip;
+
+			webClipName = webClipName.replace("webclip/", "")
```
Comment (HunseopJeong) [inline examples/mobile/HomeApp/js/app.js]: ditto

File: examples/mobile/HomeApp/js/app.js [hunk 4]
```diff
@@ -112,6 +125,58 @@
...
+			webClipName = webClipName.replace("/", "");
+
+			li.classList.add("ui-li-has-checkbox");
+			li.classList.add("ui-group-index");
+
+			input.setAttribute("type", "checkbox");
+			input.setAttribute("id", webClipName);
+
+			label.setAttribute("for", "latest-news-ckeck")
```
Comment (HunseopJeong) [inline examples/mobile/HomeApp/js/app.js]: ditto

File: examples/mobile/HomeApp/js/app.js [hunk 5]
```diff
@@ -112,6 +125,58 @@
...
+			li.appendChild(label);
+			popupList.appendChild(li);
+		});
+		tau.engine.createWidgets(popupList);
+		webClipList.forEach(function (webclip) {
+
+			if (activeWebClipList.includes(webclip)) {
+				let webClipName = webclip;
+
+				webClipName = webClipName.replace("webclip/", "")
```
Comment (HunseopJeong) [inline examples/mobile/HomeApp/js/app.js]: ditto

File: examples/mobile/HomeApp/js/app.js [hunk 6]
```diff
@@ -75,8 +75,21 @@
...
+		activeWebClipList = [];
+		webClipList.forEach(function (webclip) {
+			var webClipName = webclip;
+
+			webClipName = webClipName.replace("webclip/", "")
+			webClipName = webClipName.replace("/", "");
+			const checkbox = document.querySelector("#" + webClipName);
+
+			if (checkbox.checked)
+			{
```
Comment (HunseopJeong) [inline examples/mobile/HomeApp/js/app.js]: Opening the brace after statement.

Comment (lslachciak) [reply examples/mobile/HomeApp/js/app.js]: done

File: examples/mobile/HomeApp/js/app.js (hunk 5 above)
Comment (lslachciak) [reply examples/mobile/HomeApp/js/app.js]: done

File: examples/mobile/HomeApp/js/app.js (hunk 4 above)
Comment (lslachciak) [reply examples/mobile/HomeApp/js/app.js]: done

File: examples/mobile/HomeApp/js/app.js (hunk 3 above)
Comment (lslachciak) [reply examples/mobile/HomeApp/js/app.js]: done

File: examples/mobile/HomeApp/js/app.js (hunk 2 above)
Comment (lslachciak) [reply examples/mobile/HomeApp/js/app.js]: done

File: examples/mobile/HomeApp/js/app.js (hunk 1 above)
Comment (lslachciak) [reply examples/mobile/HomeApp/js/app.js]: done

**Comment by lslachciak** [comment]
improved according to @HunseopJeong comments.
Also removed unused structures/functions

File: examples/mobile/HomeApp/js/app.js [hunk 7]
```diff
@@ -75,8 +52,20 @@
...
 	function onPopupSubmit() {
-		getWidgetsState();
-		toggleWidgetsState();
+
+		activeWebClipList = [];
+		webClipList.forEach(function (webclip) {
+			var webClipName = webclip;
+
+			webClipName = webClipName.replace("webclip/", "");
+			webClipName = webClipName.replace("/", "");
```
Comment (korneliakobiela) [inline examples/mobile/HomeApp/js/app.js]: Isn't it cleaner to write these to expressions as a one?
``` js
webClipName = webClipName.replace("webclip/", "").replace("/", "");
```

File: examples/mobile/HomeApp/js/app.js [hunk 8]
```diff
@@ -112,6 +101,57 @@
...
+			popupList.removeChild(li);
+		});
+
+		webClipList.forEach(function (webclip) {
+			var li = document.createElement("li"),
+				input = document.createElement("input"),
+				label = document.createElement("label"),
+				webClipName = webclip;
+
+			webClipName = webClipName.replace("webclip/", "");
```
Comment (korneliakobiela) [inline examples/mobile/HomeApp/js/app.js]: Ditto. Or, better idea, encapsulate this code into the function instead of repeat the lines.

Comment (lslachciak) [reply examples/mobile/HomeApp/js/app.js]: encapsualated in function

File: examples/mobile/HomeApp/js/app.js (hunk 7 above)
Comment (lslachciak) [reply examples/mobile/HomeApp/js/app.js]: done

**Comment by lslachciak** [comment]
improved according to @korneliakobiela comments

**Review by korneliakobiela — APPROVED** [review]
Looks nice :1st_place_medal:

**Pull request merged**